sqlwhat/checks/check_funcs.py

```diff
@@ -205,8 +220,9 @@ def has_equal_ast(state, msg="Your submission is incorrect. Try again!", sql=Non
     stu_rep = repr(state.student_ast)
     sol_rep = repr(sol_ast)
 
-    if       exact and (sol_rep != stu_rep):     state.do_test(Test(msg))
-    elif not exact and (sol_rep not in stu_rep): state.do_test(Test(msg))
+    _msg = state.ast_dispatcher.describe(state.student_ast, msg, ast_path = state.get_ast_path())
```

Comment:
Here, you use `state.ast_dispatcher.describe()` and in their, you use `state.get_ast_path()`, which internally uses `state.ast_dispatcher.describe()`. Can you explain why exactly? I'm not totally following.